ipld/retriever.go

```diff
@@ -151,6 +151,19 @@ func (rs *retrievalSession) Reconstruct() (*rsmt2d.ExtendedDataSquare, error) {
 	// prevent further writes to the square
 	rs.squareLk.Lock()
 	defer rs.squareLk.Unlock()
+
+	// TODO(@Wondertan): This is bad!
+	//  * We should not reimport the square multiple times
+	//  * We should set shares into imported square via SetShare(https://github.com/celestiaorg/rsmt2d/issues/83)
```

Comment:
It's not clear to me how specifically this fixes a bug. This is indicative of a bug in the rsmt2d, not a missing feature.